ray-operator/controllers/ray/raycluster_controller_test.go

```diff
@@ -1330,4 +1331,35 @@ var _ = Context("Inside the default namespace", func() {
 				time.Second*3, time.Millisecond*500).Should(BeTrue())
 		})
 	})
+
+	Describe("RayCluster with a maximum name", func() {
+		ctx := context.Background()
+		namespace := "default"
+		rayCluster := rayClusterTemplate(strings.Repeat("r", utils.MaxRayClusterNameLength), namespace)
+		numOfHosts := int32(4)
+		rayCluster.Spec.WorkerGroupSpecs[0].NumOfHosts = numOfHosts
```

Comment:
why do we need to set `numOfHosts`?

Reply:
We need it to create the -headless service. A comment is added.